ttnn/cpp/ttnn/operations/ccl/all_gather/device/kernels/dataflow/worker_interleaved_ring_gather_receive_writer.cpp

```diff
@@ -123,6 +125,69 @@ void kernel_main() {
     uint32_t col_idx = col_start_idx;
     uint32_t row_idx = row_start_idx;
 
+#ifdef OVERLAP_OP
```

Comment:
prefer to pass in a compile time arg and enable with `if constexpr(overlap_op) {`